cmd/expiration-mailer/main_test.go

```
@@ -289,6 +290,152 @@ func TestFindExpiringCertificates(t *testing.T) {
+			IsRenewed: true,
+		},
+		{
+			Serial:    1002,
+			FQDNHash:  []byte("hash of A"),
+			DNS:       "a.example.com",
+			NotBefore: ctx.fc.Now().Add((0 * 24) * time.Hour),
+			NotAfter:  ctx.fc.Now().Add((90 * 24) * time.Hour),
+			IsRenewed: false,
+		},
+		{
+			Serial:    1003,
+			FQDNHash:  []byte("hash of B"),
+			DNS:       "b.example.net",
+			NotBefore: ctx.fc.Now().Add((0 * 24) * time.Hour),
+			NotAfter:  ctx.fc.Now().Add((90 * 24) * time.Hour),
+			IsRenewed: false,
+		},
+		{
+			Serial:    1004,
+			FQDNHash:  []byte("hash of C"),
+			DNS:       "c.example.org",
+			NotBefore: ctx.fc.Now().Add((-100 * 24) * time.Hour),
+			NotAfter:  ctx.fc.Now().Add((-10 * 24) * time.Hour),
+			IsRenewed: true,
+		},
+		{
+			Serial:    1005,
+			FQDNHash:  []byte("hash of C"),
+			DNS:       "c.example.org",
+			NotBefore: ctx.fc.Now().Add((-80 * 24) * time.Hour),
+			NotAfter:  ctx.fc.Now().Add((10 * 24) * time.Hour),
+			IsRenewed: true,
+		},
+		{
+			Serial:    1006,
+			FQDNHash:  []byte("hash of C"),
+			DNS:       "c.example.org",
+			NotBefore: ctx.fc.Now().Add((-75 * 24) * time.Hour),
+			NotAfter:  ctx.fc.Now().Add((15 * 24) * time.Hour),
+			IsRenewed: true,
+		},
+		{
+			Serial:    1007,
+			FQDNHash:  []byte("hash of C"),
+			DNS:       "c.example.org",
+			NotBefore: ctx.fc.Now().Add((-1 * 24) * time.Hour),
+			NotAfter:  ctx.fc.Now().Add((89 * 24) * time.Hour),
+			IsRenewed: false,
+		},
+	}
+
+	setupDBMap, err := sa.NewDbMap("mysql+tcp://test_setup@localhost:3306/boulder_sa_test")
+
+	for _, testData := range testCerts {
+		testData.stringSerial = strconv.Itoa(testData.Serial)
+
+		rawCert := x509.Certificate{
+			Subject: pkix.Name{
+				CommonName: testData.DNS,
+			},
+			NotBefore:    testData.NotBefore,
+			NotAfter:     testData.NotAfter,
+			DNSNames:     []string{testData.DNS},
+			SerialNumber: big.NewInt(int64(testData.Serial)),
+		}
+		certDer, _ := x509.CreateCertificate(rand.Reader, &rawCert, &rawCert, &testKey.PublicKey, &testKey)
```

We should avoid ignoring errors, even in tests. I've run into a number of situations where a test was much harder to debug because of an ignored error.